Automattic/wp-calypso
# Storage Add Ons: Implement dropdown UI functionality

Related to https://github.com/Automattic/wp-calypso/pull/79041#discussion_r1279224891
Issue https://github.com/Automattic/martech/issues/2025
Tracking https://github.com/Automattic/martech/issues/2023

DEPENDENT ON https://github.com/Automattic/wp-calypso/pull/81537

## Proposed Changes

* Update use-add-on hook to return monthly and yearly add-on costs
* Implement use add-on hook in plan-features-main component
* Add storageAddOns data to gridPlans
* Consume storage add-on data in the storage dropdown component, header pricing component, and billing timeframe component
* Only show add-on upsells in the `/start` flow

## GIFs
![2023-09-03 20 07 17](https://github.com/Automattic/wp-calypso/assets/5414230/fc88caaa-96e5-483e-bd21-fcff5200ce1b)

## Will Follow Up in Future PR
* Comparison grid storage add on dropdown width is too narrow for tablet resolutions ( longer currency values are wonky )
* Make the storage add-on in checkout cart a dropdown ( to allow user to switch between storage add-ons )

## Testing Instructions



* Smoke test signup flows with business or ecommerce plans:
  - `/start?flags=plans%2Fupgradeable-storage`
* Verify that the storage add-on dropdown only renders for business and ecommerce plans
* Verify that the dropdown is only rendered for annual plans ( not monthly )
* Verify that the header pricing and billing timeframe changes appropriately when storage add-ons are selected
* Ensure that add-on upsells do not display in other signup `/start` flows by visiting:
  - `/start/domain?flags=plans%2Fupgradeable-storage`
  - `/start/plans-import?flags=plans%2Fupgradeable-storage`
* Ensure that add-on upsells do not display in stepper flows by visiting:
  - `setup/onboarding-media?flags=plans%2Fupgradeable-storage`
* Ensure that add-on upsells do not display in `/plans` by visiting:
  - `/plans?flags=plans%2Fupgradeable-storage`

## Pre-merge Checklist
